src/controllers/util/index.ts

```diff
@@ -23,6 +28,49 @@ export default class ControllerUtility {
     [EventType.TRACK]: [`properties.${RETL_TIMESTAMP}`, ...genericFieldMap.timestamp],
   };
 
+  private static getSourceVersionsMap(): Map<string, any> {
+    if (this.sourceVersionMap.size > 0) {
```

Comment:
```suggestion
    if (this.sourceVersionMap.size && this.sourceVersionMap.size > 0) {
```

Reply:
used Optional chaining instead
this.sourceVersionMap?.size > 0